OpenLIFUData/OpenLIFUData.py

```diff
@@ -768,358 +1179,178 @@ def setup(self) -> None:
         self.setupSHNodeObserver()
         self.addObserver(slicer.mrmlScene, slicer.vtkMRMLScene.NodeAddedEvent, self.onNodeAdded)
         self.addObserver(slicer.mrmlScene, slicer.vtkMRMLScene.NodeRemovedEvent, self.onNodeRemoved)
+        
+        # Connect to the database logic for updates related to database
+        slicer.util.getModuleLogic("OpenLIFUDatabase").call_on_db_changed(self.onDatabaseChanged)
+
+        # ---- Internal button setup ----
 
         # Go to protocol config
         self.ui.configureProtocolsPushButton.clicked.connect(lambda : slicer.util.selectModule("OpenLIFUProtocolConfig"))
 
-        # Add new subject
-        self.ui.newSubjectButton.clicked.connect(self.onAddNewSubjectClicked)
-        self.update_newSubjectButton_enabled()
+        # Subject collapsible section
+        self.ui.chooseSubjectButton.clicked.connect(self.on_load_subject_clicked)
 
-        # Add new volume to subject
-        self.ui.addVolumeToSubjectButton.clicked.connect(self.onAddVolumeToSubjectClicked)
+        # Volumes collapsible section
+        self.ui.addVolumeButton.clicked.connect(self.on_add_volume_clicked)
 
-        # Add new session
-        self.ui.newSessionButton.clicked.connect(self.onCreateNewSessionClicked)
-        self.update_subjectLevelButtons_enabled()
+        # Session collapsible section
+        self.ui.chooseSessionButton.clicked.connect(self.on_load_session_clicked)
 
```

Comment:
```suggestion
self.ui.saveSessionButton.clicked.connect(self.on_save_session_clicked)
```

The button connection is missing! But I tested after adding this and the save_session logic works as expected.